h2o-py/h2o/expr.py

```diff
@@ -149,7 +150,12 @@ def _get_ast_str(self, top):
             return self._cache._id  # Data already computed under ID, but not cached
         assert isinstance(self._children,tuple)
         exec_str = "({} {})".format(self._op, " ".join([ExprNode._arg_to_expr(ast) for ast in self._children]))
-        gc_ref_cnt = len(gc.get_referrers(self))
+        # explicit call to gc.collect as recommended before calling gc.get_referrers to obtain only live referrers;
+        # otherwise, this creates subtle bugs for Py < 3.7 as we have logic below based on the ref count.
+        # Also filtering out the frames for the referrers (they appear until 3.6) and change the behaviour depending on the execution path.
+        gc.collect()
+        referrers = gc.get_referrers(self)
+        gc_ref_cnt = len([r for r in referrers if not inspect.isframe(r)])
```

Comment:
the logic below relying on gc_ref_cnt is a complete nightmare: different versions of Py don't behave the same regarding `gc`, causing very annoying bugs when switching versions.
I added the `gc.collect()` that optimizes this for Py 3.7+.
Also filtering out calling frames, as they're kept in Py 3.6-, leading to creating of  duplicate tmp frames below.

Need more work (different task) to see if we could completely remove this `MAGIC_REF_COUNT` limit.